src/org/skypro/skyshop/SearchEngine/SearchEngine.java

```diff
@@ -25,11 +25,9 @@ public void add(Searchable searchable) {
     // Метод поиска
     public Set<Searchable> search(String query) {
         // Создаем TreeSet с компаратором
-        Set<Searchable> results = new TreeSet<>(new SearchableComparator());
+        Set results = new TreeSet<>(new SearchableComparator());
```

Comment:
```suggestion
        Set<Searchable> results = new TreeSet<>(new SearchableComparator());
```
сет должен быть параметризирован